core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -337,27 +337,20 @@ public void testCatalogBasicBearerToken() {
     // the bearer token should be used for all interactions
     Mockito.verify(adapter)
         .execute(
-            eq(HTTPMethod.GET),
-            eq("v1/config"),
-            any(),
-            any(),
+            reqMatcher(HTTPMethod.GET, "v1/config", catalogHeaders),
             eq(ConfigResponse.class),
-            eq(catalogHeaders),
+            any(),
             any());
     Mockito.verify(adapter)
         .execute(
-            eq(HTTPMethod.HEAD),
-            eq("v1/namespaces/ns/tables/table"),
+            reqMatcher(HTTPMethod.HEAD, "v1/namespaces/ns/tables/table", catalogHeaders),
             any(),
             any(),
-            any(),
-            eq(catalogHeaders),
             any());
   }
 
   @Test
   public void testCatalogCredentialNoOauth2ServerUri() {
-    Map<String, String> emptyHeaders = ImmutableMap.of();
```

Comment:
nit: we might want to consider keeping this so that it's clearer that empty headers are expected